File: Backend/connecthub/src/main/java/com/example/connecthub/service/ConnectionService.java
package com.example.connecthub.service;

import com.example.connecthub.dto.response.ConnectionResponse;
import com.example.connecthub.dto.response.ConnectionUserResponse;

import java.util.List;

public interface ConnectionService {

    ConnectionResponse sendConnectionRequest(
            String email,
            Long receiverId
    );

    List<ConnectionUserResponse> getReceivedRequests(
            String email
    );

    List<ConnectionUserResponse> getSentRequests(
            String email
    );

    List<ConnectionUserResponse> getConnections(
            String email
    );

    ConnectionResponse acceptConnection(
            Long connectionId,
            String email
    );

    ConnectionResponse rejectConnection(
            Long connectionId,
            String email
    );

    void removeConnection(
            Long connectionId,
            String email
    );
}

File: frontend/src/services/connectionService.js
import client from '../api/client';

export const connectionService = {
  getConnections: async () => {
    const response = await client.get('/connections');
    return response.data; // List<ConnectionUserResponse>
  },

  getReceivedRequests: async () => {
    const response = await client.get('/connections/received');
    return response.data; // List<ConnectionUserResponse>
  },

  getSentRequests: async () => {
    const response = await client.get('/connections/sent');
    return response.data; // List<ConnectionUserResponse>
  },

  sendConnectionRequest: async (userId) => {
    const response = await client.post(`/connections/${userId}`);
    return response.data; // ConnectionResponse
  },

  acceptConnection: async (connectionId) => {
    const response = await client.put(`/connections/${connectionId}/accept`);
    return response.data; // ConnectionResponse
  },

  rejectConnection: async (connectionId) => {
    const response = await client.put(`/connections/${connectionId}/reject`);
    return response.data; // ConnectionResponse
  },

  removeConnection: async (connectionId) => {
    await client.delete(`/connections/${connectionId}`);
  },
};
